github_app.go

```diff
@@ -171,17 +171,19 @@ func (s *Settings) CreateInstallationToken(ctx context.Context, owner, repo stri
 			}))
 			return "", errors.Wrapf(err, "finding installation id for '%s/%s'", owner, repo)
 		}
-		if installationID == nil {
+		if installation == nil {
 			return "", errors.Errorf("Installation id for '%s/%s' not found", owner, repo)
 		}
 
+		installationID = installation.GetID()
+
 		// Cache the installation ID for owner/repo.
-		installation := GitHubAppInstallation{
+		cachedInstallation := GitHubAppInstallation{
 			Owner:          owner,
 			Repo:           repo,
-			InstallationID: installationID.GetID(),
+			InstallationID: installationID,
```

Comment:
How is this different from the previous code where we were just assigning `InstallationID: installationID.GetID(),` directly in the struct?

Reply:
talked offline but essentially installationID was being defined twice. Once inside the if and once outside, so changing it on the inside did not do anything. Changing the names solves this problem